File: .tmp_py4013/batch_17_dump/3853_merge_close_characters.js
// LeetCode 3853 - Merge Close Characters
// https://leetcode.com/problems/merge-close-characters/

var mergeCharacters = function(s, k) {
    const last = new Map();
    let ans = '';
    for (const c of s) {
        const cur = ans.length;
        if (last.has(c) && cur - last.get(c) <= k) continue;
        ans += c;
        last.set(c, cur);
    }
    return ans;
};

File: .tmp_rs4013/rust_check21/3853_merge_close_characters.rs
struct Solution;
// LeetCode 3853 - Merge Close Characters
// https://leetcode.com/problems/merge-close-characters/

use std::collections::HashMap;

impl Solution {
    pub fn merge_characters(s: String, k: i32) -> String {
        let mut last = HashMap::new();
        let mut ans = String::new();
        for c in s.chars() {
            let cur = ans.len() as i32;
            if let Some(&prev) = last.get(&c) {
                if cur - prev <= k {
                    continue;
                }
            }
            ans.push(c);
            last.insert(c, cur);
        }
        ans
    }
}
